sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3510,4 +3510,34 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
       errorClass = "CANNOT_RENAME_ACROSS_SCHEMA", messageParameters = Map("type" -> "table")
     )
   }
+
+  def avroIncorrectTypeError(
+      avroPath: String, sqlPath: String, avroType: String,
+      sqlType: String, key: String): Throwable = {
+    new AnalysisException(
```

Comment:
should these be AnalysisExceptions? I feel it should be SparkRuntimeExceptions

Reply:
I am not 100% sure about it. @gengliangwang should it be `SparkRuntimeExceptions`?

Reply:
Yes it should be AnalysisExceptions

Reply:
Sure, thanks!